docker/build_image.sh

```diff
@@ -7,6 +7,8 @@ BRANCH_NAME="master"
 DOCKER_TAG="pytorch/torchserve:latest-cpu"
 BUILD_TYPE="production"
 BASE_IMAGE="ubuntu:20.04"
+USER_BASE_IMAGE="ubuntu:20.04"
```

Comment:
it is kind of confusing. Can we just use base_image (default ubuntu:20.04), and allow user input to overwrite it?

Reply:
@lxning I would appreciate your inputs if you know a good way to do this.

The problem with using just `BASE_IMAGE` is that depending on the order of `-bi` and `-g`, the base image is different. The second option will overwrite `BASE_IMAGE`

```
(torchserve) ubuntu@ip-172-31-60-100:~/serve/docker$ ./build_image.sh -g -bi nvidia/cuda:11.7.1-base-ubuntu20.04 -t py:test1
Base Image is  nvidia/cuda:11.7.1-base-ubuntu20.04
```

```
(torchserve) ubuntu@ip-172-31-60-100:~/serve/docker$ ./build_image.sh -bi nvidia/cuda:11.7.1-base-ubuntu20.04  -g -t py:test1                                                             
Base Image is  nvidia/cuda:11.8.0-base-ubuntu20.04                                                                      
```

Reply:
@lxning Changed the variable to `OVERRIDE_BASE_IMAGE` . This should make it clear

Reply:
Can we avoid the issue of ordering? For example.

Warn user "bi" input is ignored if "-g" is chosen

Reply:
@lxning Updated the code.  Print and error and exit the script